python/gui/auto_generated/vector/qgsvectorlayerproperties.sip.in

```diff
@@ -0,0 +1,120 @@
+/************************************************************************
+ * This file has been generated automatically from                      *
+ *                                                                      *
+ * src/gui/vector/qgsvectorlayerproperties.h                            *
+ *                                                                      *
+ * Do not edit manually ! Edit header and run scripts/sipify.pl again   *
+ ************************************************************************/
+
+
+
+
+
+
+class QgsVectorLayerProperties : QgsOptionsDialogBase
+{
+%Docstring
+*************************************************************************
+
+This program is free software; you can redistribute it and/or modify  *
+it under the terms of the GNU General Public License as published by  *
+the Free Software Foundation; either version 2 of the License, or     *
+(at your option) any later version.                                   *
+
+**************************************************************************
+%End
+
+%TypeHeaderCode
+#include "qgsvectorlayerproperties.h"
+%End
+  public:
+    enum StyleType
+    {
+      QML,
+      SLD,
+      DB,
+    };
+
+    QgsVectorLayerProperties( QgsMapCanvas *canvas, QgsMessageBar *messageBar, QgsVectorLayer *lyr = 0, QWidget *parent = 0, Qt::WindowFlags fl = QgsGuiUtils::ModalDialogFlags );
+
+    QString displayName();
+%Docstring
+Returns the display name entered in the dialog
+%End
+
+    void addPropertiesPageFactory( QgsMapLayerConfigWidgetFactory *factory );
+%Docstring
+Adds a properties page factory to the vector layer properties dialog.
+%End
+
+  public slots:
```

Comment:
There's a lot exposed here which shouldn't be -- I'd remove everything from the sip bindings except for the constructor.